source/common/router/config_impl.cc

```diff
@@ -977,6 +979,9 @@ RouteMatcher::RouteMatcher(const envoy::api::v2::RouteConfiguration& route_confi
         default_virtual_host_ = virtual_host;
       } else if (domain.size() > 0 && '*' == domain[0]) {
         wildcard_virtual_host_suffixes_[domain.size() - 1].emplace(domain.substr(1), virtual_host);
+      } else if (domain.size() > 0 && '*' == domain[domain.size() - 1]) {
+        wildcard_virtual_host_prefixes_[domain.size() - 1].emplace(
+            domain.substr(0, domain.size() - 1), virtual_host);
```

Comment:
Same with the suffixes above. But I just noticed that we don't detect duplicates (at least afaict here) for suffix or prefix domains, like we do for exact matches below. Is that intentional?

Reply:
I don't personally think it was intentional. I'll add duplicate checking to make things clear.